#! /bin/sh /usr/share/dpatch/dpatch-run
## 16-fix-pidl-install.dpatch by Thomas Girard <thomas.g.girard@free.fr>
##
## DP: Adds missing .pidl files to make install

@DPATCH@
--- ACE_wrappers.orig/TAO/tao/BiDir_GIOP/BiDir_GIOP.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/BiDir_GIOP/BiDir_GIOP.mpc	2011-01-05 23:14:40.610413177 +0100
@@ -1,5 +1,5 @@
 //$Id: BiDir_GIOP.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*idl) : tao_versioning_idl_defaults {
+project(*idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -Gp -Gd -Sci -SS -Sorb -Sa -St \
@@ -10,6 +10,10 @@
                 -iC BiDir_GIOP
     BiDirPolicy.pidl
   }
+
+  specific {
+    install_dir = tao/BiDir_GIOP
+  }
 }
 
 project(BiDir_GIOP) : taolib, tao_output, install, pi, taoidldefaults {
--- ACE_wrappers.orig/TAO/tao/Codeset/Codeset.mpc	2010-10-12 16:18:45.000000000 +0200
+++ ACE_wrappers/TAO/tao/Codeset/Codeset.mpc	2011-01-05 23:16:55.762564847 +0100
@@ -1,5 +1,5 @@
 //$Id: Codeset.mpc 92205 2010-10-12 14:18:45Z mitza $
-project(*idl) : tao_versioning_idl_defaults {
+project(*idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   idlflags += -SS -Sci -Sorb \
               -Wb,export_macro=TAO_Codeset_Export \
@@ -8,6 +8,10 @@
   IDL_Files {
     CodeSetContext.pidl
   }
+
+  specific {
+    install_dir = tao/Codeset
+  }
 }
 
 project(Codeset) : taolib, tao_output, install, taoidldefaults, nolink_codecfactory {
--- ACE_wrappers.orig/TAO/tao/CSD_Framework/CSD_Framework.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/CSD_Framework/CSD_Framework.mpc	2011-01-05 23:17:19.530568257 +0100
@@ -1,5 +1,5 @@
 //$Id: CSD_Framework.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*idl) : tao_versioning_idl_defaults {
+project(*idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -Sci -SS -Gp -Gd -Sorb -GA -Sal \
@@ -9,6 +9,10 @@
     idlflags -= -Sa -St
     CSD_Framework.pidl
   }
+
+  specific {
+    install_dir = tao/CSD_Framework
+  }
 }
 
 project(CSD_Framework) : taolib, tao_output, portableserver, install, pi, avoids_corba_e_micro, taoidldefaults {
--- ACE_wrappers.orig/TAO/tao/DiffServPolicy/DiffServPolicy.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/DiffServPolicy/DiffServPolicy.mpc	2011-01-05 23:17:41.642430721 +0100
@@ -1,5 +1,5 @@
 //$Id: DiffServPolicy.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*idl) : tao_versioning_idl_defaults, tao_no_iiop {
+project(*idl) : tao_versioning_idl_defaults, tao_no_iiop, install {
   custom_only = 1
   IDL_Files {
     idlflags += -GA -SS -Sci -Sorb \
@@ -11,6 +11,10 @@
     idlflags -= -St
     DiffServPolicy.pidl
   }
+
+  specific {
+    install_dir = tao/DiffServPolicy
+  }
 }
 
 project(DiffServPolicy) : taolib, tao_output, install, portableserver, pi, taoidldefaults, tao_no_iiop, avoids_minimum_corba, avoids_corba_e_compact, avoids_corba_e_micro {
--- ACE_wrappers.orig/TAO/tao/EndpointPolicy/EndpointPolicy.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/EndpointPolicy/EndpointPolicy.mpc	2011-01-05 23:18:10.082574034 +0100
@@ -1,5 +1,5 @@
 //$Id: EndpointPolicy.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*core_idl) : tao_versioning_idl_defaults {
+project(*core_idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -GA -SS -Sci -Sorb \
@@ -11,6 +11,10 @@
     IIOPEndpointValue.pidl
     EndpointPolicyType.pidl
   }
+
+  specific {
+    install_dir = tao/EndpointPolicy
+  }
 }
 
 project(EndpointPolicy) : taolib, tao_output, install, portableserver, pi, taoidldefaults, tao_no_iiop, avoids_minimum_corba, avoids_corba_e_compact, avoids_corba_e_micro {
--- ACE_wrappers.orig/TAO/tao/DynamicAny/DynamicAny.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/DynamicAny/DynamicAny.mpc	2011-01-05 23:18:44.986574465 +0100
@@ -1,5 +1,5 @@
 //$Id: DynamicAny.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*idl) : tao_versioning_idl_defaults, valuetype {
+project(*idl) : tao_versioning_idl_defaults, valuetype, install {
   custom_only = 1
   IDL_Files {
     idlflags += -Gp -Gd -Sci -SS -Sorb \
@@ -10,6 +10,10 @@
                 -iC tao/DynamicAny
     DynamicAny.pidl
   }
+
+  specific {
+    install_dir = tao/DynamicAny
+  }
 }
 
 project(DynamicAny) : install, tao_output, valuetype, taoidldefaults {
--- ACE_wrappers.orig/TAO/tao/ImR_Client/ImR_Client.mpc	2010-09-20 17:28:42.000000000 +0200
+++ ACE_wrappers/TAO/tao/ImR_Client/ImR_Client.mpc	2011-01-05 23:19:11.594581912 +0100
@@ -1,5 +1,5 @@
 //$Id: ImR_Client.mpc 91869 2010-09-20 15:28:42Z mitza $
-project(*idl) : tao_versioning_idl_defaults {
+project(*idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -Gp -Ssi -Gd -Sorb -GA \
@@ -18,6 +18,10 @@
     idlflags -= -Sa -St
     ImplRepo.pidl
   }
+
+  specific {
+    install_dir = tao/ImR_Client
+  }
 }
 
 project(ImR_Client) : taolib, tao_output, install, portableserver, taoidldefaults {
--- ACE_wrappers.orig/TAO/tao/IORInterceptor/IORInterceptor.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/IORInterceptor/IORInterceptor.mpc	2011-01-05 23:20:40.282592889 +0100
@@ -1,5 +1,5 @@
 //$Id: IORInterceptor.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*idl) : tao_versioning_idl_defaults {
+project(*idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -Gp -Gd -Sci -SS -Sorb -Sa -St \
@@ -18,6 +18,10 @@
                 -iC IORInterceptor
     IORInfo.pidl
   }
+
+  specific {
+    install_dir = tao/IORInterceptor
+  }
 }
 
 project(IORInterceptor) : taolib, tao_output, portableserver, objreftemplate, pi, install, taoidldefaults {
--- ACE_wrappers.orig/TAO/tao/IORManipulation/IORManipulation.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/IORManipulation/IORManipulation.mpc	2011-01-05 23:19:59.674587812 +0100
@@ -1,5 +1,5 @@
 //$Id: IORManipulation.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*idl) : tao_versioning_idl_defaults {
+project(*idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -Gp -Gd -Sci -SS -Sorb -GX \
@@ -11,6 +11,10 @@
     idlflags -= -Sa -St
     IOR.pidl
   }
+
+  specific {
+    install_dir = tao/IORManipulation
+  }
 }
 
 project(IORManipulation) : taolib, tao_output, install, anytypecode, taoidldefaults {
--- ACE_wrappers.orig/TAO/tao/IORTable/IORTable.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/IORTable/IORTable.mpc	2011-01-05 23:20:15.906590207 +0100
@@ -1,5 +1,5 @@
 //$Id: IORTable.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*core_idl) : tao_versioning_idl_defaults {
+project(*core_idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -Gp -Gd -Sci -SS -Sorb \
@@ -10,6 +10,10 @@
                 -iC tao/IORTable
     IORTable.pidl
   }
+
+  specific {
+    install_dir = tao/IORTable
+  }
 }
 
 project(IORTable) : taolib, tao_output, install, taoidldefaults {
--- ACE_wrappers.orig/TAO/tao/Strategies/Strategies.mpc	2010-09-18 00:20:25.000000000 +0200
+++ ACE_wrappers/TAO/tao/Strategies/Strategies.mpc	2011-01-05 23:21:29.298599747 +0100
@@ -1,5 +1,5 @@
 //$Id: Strategies.mpc 91842 2010-09-17 22:20:25Z mitza $
-project(*core_idl) : tao_versioning_idl_defaults {
+project(*core_idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -Sci -SS -Gp -Gd -Sorb -DCORBA3 \
@@ -12,6 +12,9 @@
     COIOP_Endpoints.pidl
   }
 
+  specific {
+    install_dir = tao/Strategies
+  }
 }
 
 project(Strategies) : taolib, tao_output, install, pi, anytypecode, core_strategies, taoidldefaults {
--- ACE_wrappers.orig/TAO/tao/TransportCurrent/TC.mpc	2010-09-20 16:49:07.000000000 +0200
+++ ACE_wrappers/TAO/tao/TransportCurrent/TC.mpc	2011-01-05 23:22:23.674594865 +0100
@@ -1,5 +1,5 @@
 //$Id: TC.mpc 91868 2010-09-20 14:49:07Z mitza $
-project(*idl) : taolib_with_idl, tao_versioning_idl_defaults {
+project(*idl) : taolib_with_idl, tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -Wb,export_include=tao/TransportCurrent/Transport_Current_Export.h \
@@ -9,6 +9,10 @@
 
     TC.idl
   }
+
+  specific {
+    install_dir = tao/TransportCurrent
+  }
 }
 
 project(TC) : taolib_with_idl, tao_output, taoidldefaults, install, interceptors, pi {
--- ACE_wrappers.orig/TAO/tao/TransportCurrent/TC_IIOP.mpc	2010-09-20 16:49:07.000000000 +0200
+++ ACE_wrappers/TAO/tao/TransportCurrent/TC_IIOP.mpc	2011-01-05 23:23:15.466612995 +0100
@@ -1,5 +1,5 @@
 //$Id: TC_IIOP.mpc 91868 2010-09-20 14:49:07Z mitza $
-project(*idl) : tao_versioning_idl_defaults {
+project(*idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -Wb,export_include=tao/TransportCurrent/Transport_Current_Export.h \
@@ -9,6 +9,10 @@
 
     TC_IIOP.idl
   }
+
+  specific {
+    install_dir = tao/TransportCurrent
+  }
 }
 
 project(TC_IIOP) : taolib_with_idl, tao_output, taoidldefaults, install, tc, interceptors, pi {
--- ACE_wrappers.orig/TAO/tao/TypeCodeFactory/TypeCodeFactory.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/TypeCodeFactory/TypeCodeFactory.mpc	2011-01-05 23:29:05.058441563 +0100
@@ -1,5 +1,5 @@
 //$Id: TypeCodeFactory.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*idl) : tao_versioning_idl_defaults {
+project(*idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -SS -Sci -Sorb \
@@ -9,6 +9,10 @@
     idlflags -= -Sa -St
     TypeCodeFactory.pidl
   }
+
+  specific {
+    install_dir = tao/TypeCodeFactory
+  }
 }
 
 project(TypeCodeFactory) : taolib, tao_output, install, ifr_client, taoidldefaults {
--- ACE_wrappers.orig/TAO/tao/Valuetype/Valuetype.mpc	2010-09-18 00:20:25.000000000 +0200
+++ ACE_wrappers/TAO/tao/Valuetype/Valuetype.mpc	2011-01-05 23:30:09.194666147 +0100
@@ -1,5 +1,5 @@
 //$Id: Valuetype.mpc 91842 2010-09-17 22:20:25Z mitza $
-project(*idl) : tao_versioning_idl_defaults, gen_ostream {
+project(*idl) : tao_versioning_idl_defaults, gen_ostream, install {
   custom_only = 1
   IDL_Files {
     idlflags += -SS -Sorb \
@@ -9,6 +9,10 @@
     idlflags -= -St
     StringValue.pidl
   }
+
+  specific {
+    install_dir = tao/Valuetype
+  }
 }
 
 project(Valuetype) : taolib, tao_output, install, anytypecode, avoids_corba_e_micro, taoidldefaults, gen_ostream, valuetype_out_indirection {
--- ACE_wrappers.orig/TAO/tao/ZIOP/ZIOP.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/ZIOP/ZIOP.mpc	2011-01-05 23:31:03.082440988 +0100
@@ -1,5 +1,5 @@
 //$Id: ZIOP.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*core_idl) : tao_versioning_idl_defaults {
+project(*core_idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags -= -Sa -St
@@ -11,6 +11,10 @@
                 -iC tao/ZIOP
     ZIOP.pidl
   }
+
+  specific {
+    install_dir = tao/ZIOP
+  }
 }
 
 project(ZIOP) : tao_output, install, taolib, compression, pi, taoidldefaults {
--- ACE_wrappers.orig/TAO/tao/tao.mpc	2010-08-27 10:16:09.000000000 +0200
+++ ACE_wrappers/TAO/tao/tao.mpc	2011-01-05 23:34:03.610695195 +0100
@@ -55,6 +55,8 @@
   }
 
   PidlInstallWithoutBuilding_Files {
+    InterfaceDef.pidl
+    InvalidName.pidl
     Object_Key.pidl
     Policy.pidl
     Typecode_types.pidl
--- ACE_wrappers.orig/TAO/tao/Messaging/Messaging.mpc	2010-09-17 22:23:03.000000000 +0200
+++ ACE_wrappers/TAO/tao/Messaging/Messaging.mpc	2011-01-05 23:37:36.682722156 +0100
@@ -1,5 +1,5 @@
 //$Id: Messaging.mpc 91840 2010-09-17 20:23:03Z mitza $
-project(*idl) : tao_versioning_idl_defaults {
+project(*idl) : tao_versioning_idl_defaults, install {
   custom_only = 1
   IDL_Files {
     idlflags += -SS -GA -Gp -Gd -Sorb -Sal \
@@ -60,6 +60,10 @@
     idlflags -= -Sa -St
     Messaging.pidl
   }
+
+  specific {
+    install_dir = tao/Messaging
+  }
 }
 
 project(Messaging) : taolib, tao_output, install, portableserver, pi, valuetype, taoidldefaults {
